js/app.jsx

```diff
@@ -297,101 +416,94 @@ var CaseverListItem = React.createClass({
-    }.bind(this))
+    var obj = this.props.setting.header;
+    var tableheader = obj.map(
+      function(setting){
+        if (setting["sortable"]) {
+          return <SearchTableHeadTh
+            text={setting["text"]}
+            filter={setting["filtername"]}
+            handleAddFilter={this.props.handleAddFilter} />
+        } else {
+          return <SearchTableHeadConstTh
+            text={setting["text"]} />
+        }
+      }.bind(this))
 
     return (
-      <Row>
-      <Table striped condensed hover className="caseverList">
-        <tbody>
-          <tr>
-            <th></th>
-            <th>ID</th>
-            <th>status</th>
-            <SortableTh name="name" filter="name" handleAddFilter={this.props.handleAddFilter}></SortableTh>
-            <SortableTh name="priority" filter="case__priority" handleAddFilter={this.props.handleAddFilter}></SortableTh>
-            <SortableTh name="product" filter="productversion" handleAddFilter={this.props.handleAddFilter}></SortableTh>
-            <SortableTh name="modified" filter="modified_on" handleAddFilter={this.props.handleAddFilter}></SortableTh>
-            <th></th>
-            <th></th>
-          </tr>
-          {casevers}
-        </tbody>
-      </Table>
-      </Row>
+      <tr>{tableheader}</tr>
     )
   }
-});
-
-var SearchableCaseverList = React.createClass({
-  mixins: [SearchableRemoteListMixin],
-  api_url: config.baseUrl + "/api/v1/caseversion/",
-  //TODO: migrate to api_url: "https://moztrap.mozilla.org/api/v1/caseversionsearch/",
-  buildURL: function(query) {
-      return (buildQueryUrl(this.api_url, query, caseversionCodegen) + 
-              "&limit=" + config.defaultListLimit
-              //"&order_by=" + "-modified_on"
-             );
-  },
+})
 
-  handleQueueUpdate: function(e) {
-    if (e.target.checked){
-      var newState = {};
-      newState['checked'] = this.state['checked'].concat(e.target.value);
-      //console.log('will set state')
-      this.setState(newState);
-    }
-    else {
-      this.state['checked'].splice(this.state['checked'].indexOf(e.target.value), 1);
-      var newState = {};
-      newState['checked'] = this.state['checked'];
-      //console.log('will set state')
-      this.setState(newState);
-    }
-  },
-  /*
-  diff: function(){
-    alert(this.state.checked)
-  },
-  */
+var SearchTable = React.createClass({
   render: function() {
-    //update
-    //
-    var diffURL = ""
-    var diffDisabled = true;
-    if (typeof this.state.checked !== "undefined"){
-      console.log("this.state is not undefined")
-      diffURL = "diff.html?lhs=" + this.state.checked[0] + "&rhs=" + this.state.checked[1]
-      if (this.state.checked.length == 2){
-        var diffDisabled = false;
-      }
+    var content
+    if (this.props.setting.pagename == "caseversion") {
+      content = this.props.data.objects.map(
+          function(casever) {
+            return (<CaseverListItem
```

Comment:
I think it can be done. But it raise another problem, the data of `casever` and `suite` is different: one is `casever` and one is `suite`. We need to make them same before unify this class creation.